### PR TITLE
GrafanaUI: change active tab styles to prevent layout shift

### PR DESCRIPTION
**Pull request by going-confetti**
**What is this feature?**
This PR removes unifies `font-weight` across different tab states. It also replaces the deprecated `stylesFactory` with `useStyles2`

**Why do we need this feature?**
Since currently active tabs have different font-weight, switching between tabs (often) leads to a layout shift. Basically, the longer the label - the bigger the difference in widths between font-weight: 400 and 500 - the easier it is to notice the shift.

Before

https://user-images.githubusercontent.com/3773351/222712977-a250b505-8a1f-489c-9eb4-45e4ba4f0580.mov

After

https://user-images.githubusercontent.com/3773351/222713034-a81cd23c-544f-4e82-9c3e-75076e4fc40b.mov

**Who is this feature for?**
All users of Grafana

### CODE REVIEW

File: packages/grafana-ui/src/components/Tabs/Tab.tsx [hunk 1]
```diff
@@ -73,12 +72,24 @@ const getTabStyles = stylesFactory((theme: GrafanaTheme2) => {
...
       svg {
         margin-right: ${theme.spacing(1)};
       }
 
       &:focus-visible {
-+        ${getFocusStyles(theme)}
+        ${getFocusStyles(theme)}
+      }
+
+      &::before {
```
Comment (going-confetti) [inline packages/grafana-ui/src/components/Tabs/Tab.tsx:84]: The idea behind this change is to remove code duplication and only change the background depending on the tab state.

File: packages/grafana-ui/src/components/Tabs/Tab.tsx [hunk 2]
```diff
@@ -88,42 +99,25 @@ const getTabStyles = stylesFactory((theme: GrafanaTheme2) => {
...
-          background: ${theme.colors.action.hover};
+          background-color: ${theme.colors.action.hover};
         }
       }
     `,
     activeStyle: css`
       label: activeTabStyle;
       color: ${theme.colors.text.primary};
       overflow: hidden;
-      font-weight: ${theme.typography.fontWeightMedium};
```
Comment (going-confetti) [inline packages/grafana-ui/src/components/Tabs/Tab.tsx:107]: This is the fix itself

File: packages/grafana-ui/src/components/Tabs/Tab.tsx [hunk 3]
```diff
@@ -61,7 +60,7 @@ export const Tab = React.forwardRef<HTMLAnchorElement, TabProps>(
 
 Tab.displayName = 'Tab';
 
-const getTabStyles = stylesFactory((theme: GrafanaTheme2) => {
+const getStyles = (theme: GrafanaTheme2) => {
```
Comment (going-confetti) [inline packages/grafana-ui/src/components/Tabs/Tab.tsx:63]: There's something I found confusing in the styles here: we apply some styling rules to `a` that are nested in `.link` on lines 96 and 111 (85 and 109 in `main`) and I'm not sure that's valid. Unless someone passes an anchor element to the `suffix` prop, but an an anchor element [shouldn't be a descendant of another anchor element](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element).

File: packages/grafana-ui/src/components/Tabs/Tab.tsx (hunk 2 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/Tabs/Tab.tsx:107]: The other option would be to have all be 500 weight

Comment (going-confetti) [reply packages/grafana-ui/src/components/Tabs/Tab.tsx:107]: True, didn't think of that option. Here are screenshots for comparison:
image
image

Comment (going-confetti) [reply packages/grafana-ui/src/components/Tabs/Tab.tsx:108]: I decided to use 400 as that's what we have in Figma and newer core components like [SectionNavItem](https://github.com/grafana/grafana/blob/7860ca6c3d7060adb7af33b5d08ad5926fcaecaf/public/app/core/components/PageNew/SectionNavItem.tsx#L84)

**Review by JoaoSilvaGrafana — APPROVED** [review]
Lgtm!

**Pull request merged**